Added environment creation and doc tools install (#72)
napari/docs: docs/developers/contributing.md

In order to make changes to `napari`, you will need to [fork](https://docs.github.com/en/get-started/quickstart/contributing-to-projects#forking-a-repository) the
[repository](https://github.com/napari/napari). If you are not familiar with `git`, we recommend reading up on [this guide](https://docs.github.com/en/get-started/using-git/about-git#basic-git-commands).

1. Clone the forked repository to your local machine and change directories:

    ```sh
    git clone https://github.com/your-username/napari.git
    cd napari
    ```

2. Set the `upstream` remote to the base `napari` repository:

    ```sh
    git remote add upstream https://github.com/napari/napari.git
    ```

3. If you haven't already, create a development environment:

    ::::{tab-set}

    :::{tab-item} Using `venv`
    After installing Python on your machine, create a virtual environment on your terminal and activate it. On Linux and MacOS, you can run
    ```sh
    python -m venv <path-to-env>
    source <path-to-env>/bin/activate
    ```
    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
    :::

    :::{tab-item} Using `conda`

>     After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` with Python 3.9 and activate it.

Reviewer: ```suggestion
    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env`—here with Python 3.9—and activate it.
```

    ```sh
    conda create -y --name napari-env -c conda-forge python=3.9
    conda activate napari-env
    ```
    :::

    ::::

    ```{note}
    It is highly recommended to create a fresh environment when working with
    napari, to prevent issues with outdated or conflicting packages in your
    development environment.
    ```

4. Install the package in editable mode, along with all of the developer tools.

    If you only want to use napari, you can install it on most macOS, Linux and
    Windows systems with

>     ![Supported Python versions](https://img.shields.io/pypi/pyversions/napari.svg)

Reviewer: I personally don't like the look of the shields in-line with text:
<img width="754" alt="image" src="https://user-images.githubusercontent.com/76622105/209573533-abf9a513-c58b-44f3-b64f-5214c0a5d197.png">

But it's not a blocker for me.

    by following the directions on the

>     [instructions page](installation#install-as-python-package-recommended).

Reviewer: ```suggestion
    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).
```
For me the current link is missing, I think this should be correct

Reviewer: Not sure I wouldn't just remove this paragraph about the standard installation anyways.

    To install the package in editable mode for development, along with all of
    the developer tools, use the following command:
    ```sh
    pip install -e ".[dev]"  # (quotes only needed for zsh shell)
    ```

> 5. We use [`pre-commit`](https://pre-commit.com) to sort imports with
>    [`isort`](https://github.com/PyCQA/isort), format code with
>    [`black`](https://github.com/psf/black), and lint with
>    [`flake8`](https://github.com/PyCQA/flake8) automatically prior to each commit.

Reviewer: ```suggestion
5. We use [`pre-commit`](https://pre-commit.com) to identify and fix some style issues automatically prior to each commit. Currently, this includes sorting imports with
   [`isort`](https://github.com/PyCQA/isort), formating code with
   [`black`](https://github.com/psf/black), and linting with
   [`flake8`](https://github.com/PyCQA/flake8).
```

In order to make changes to `napari`, you will need to [fork](https://docs.github.com/en/get-started/quickstart/contributing-to-projects#forking-a-repository) the
[repository](https://github.com/napari/napari). If you are not familiar with `git`, we recommend reading up on [this guide](https://docs.github.com/en/get-started/using-git/about-git#basic-git-commands).

1. Clone the forked repository to your local machine and change directories:

    ```sh
    git clone https://github.com/your-username/napari.git
    cd napari
    ```

2. Set the `upstream` remote to the base `napari` repository:

    ```sh
    git remote add upstream https://github.com/napari/napari.git
    ```

3. If you haven't already, create a development environment:

    ::::{tab-set}

    :::{tab-item} Using `venv`
    After installing Python on your machine, create a virtual environment on your terminal and activate it. On Linux and MacOS, you can run
    ```sh
    python -m venv <path-to-env>
    source <path-to-env>/bin/activate
    ```
    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
    :::

    :::{tab-item} Using `conda`

>     After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` - here with Python 3.9 - and activate it.

Reviewer: ```suggestion
    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` - here with Python {{ python_version }} - and activate it.
```

>     ```sh
>     conda create -y --name napari-env -c conda-forge python=3.9
>     conda activate napari-env
>     ```

Reviewer: ```suggestion
    {{ conda_create_env }}
```

    :::

    ::::

    ```{note}
    It is highly recommended to create a fresh environment when working with
    napari, to prevent issues with outdated or conflicting packages in your
    development environment.
    ```

4. Install the package in editable mode, along with all of the developer tools.

    If you only want to use napari, you can install it on most macOS, Linux and
    Windows systems with

>     ![Supported Python versions](https://img.shields.io/pypi/pyversions/napari.svg)

Reviewer: ```suggestion
    Python {{ python_version_range }}
```

    by following the directions on the
    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).

>     To install the package in editable mode for development, along with all of

Reviewer: ```suggestion
    To install the package in editable mode for development, along with PyQt5 and all of
```

    the developer tools, use the following command:
    ```sh

>     pip install -e ".[dev]"  # (quotes only needed for zsh shell)

Reviewer: ```suggestion
    pip install -e ".[pyqt,dev]"  # (quotes only needed for zsh shell)
```

## Setting up a development installation

In order to make changes to `napari`, you will need to [fork](https://docs.github.com/en/get-started/quickstart/contributing-to-projects#forking-a-repository) the
[repository](https://github.com/napari/napari). If you are not familiar with `git`, we recommend reading up on [this guide](https://docs.github.com/en/get-started/using-git/about-git#basic-git-commands).

1. Clone the forked repository to your local machine and change directories:

    ```sh
    git clone https://github.com/your-username/napari.git
    cd napari
    ```

2. Set the `upstream` remote to the base `napari` repository:

    ```sh
    git remote add upstream https://github.com/napari/napari.git
    ```

3. If you haven't already, create a development environment:

    ::::{tab-set}

    :::{tab-item} Using `venv`
    After installing Python on your machine, create a virtual environment on your terminal and activate it. On Linux and MacOS, you can run
    ```sh
    python -m venv <path-to-env>
    source <path-to-env>/bin/activate
    ```
    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
    :::

>     :::{tab-item} Using `conda`

Reviewer 2: I wonder if we should put the `conda` instructions first? I think it's far more commonly used, and then for stuff like the arm64 mac stuff you basically have to use `conda` if I understand correctly?

Author: Done

    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` with Python {{ python_version }} and activate it.

    {{ conda_create_env }}
    :::

    ::::

    ```{note}
    It is highly recommended to create a fresh environment when working with
    napari, to prevent issues with outdated or conflicting packages in your
    development environment.
    ```

4. Install the package in editable mode, along with all of the developer tools.

    If you only want to use napari, you can install it on most macOS, Linux and
    Windows systems with Python {{ python_version_range }}
    by following the directions on the
    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).

    napari supports different Qt backends, and you can choose which one to install and use.

    For example, for PyQt5, the default, you would use the following:
    ```sh
    pip install -e ".[pyqt,dev]"  # (quotes only needed for zsh shell)
    ```

    If you want to use PySide2 instead, you would use:
    ```sh
    pip install -e ".[pyside,dev]"  # (quotes only needed for zsh shell)
    ```

>     Finally, if you already have a Qt backend installed or want to use an experimental one like Qt6 use:

Reviewer 2: Would be good to add a small note or comment here that you will then also need to manually install your backend. I know it's obvious maybe but still

Author: Let me know if this is enough

    napari, to prevent issues with outdated or conflicting packages in your
    development environment.
    ```

4. Install the package in editable mode, along with all of the developer tools.

    If you only want to use napari, you can install it on most macOS, Linux and
    Windows systems with Python {{ python_version_range }}
    by following the directions on the
    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).

    napari supports different Qt backends, and you can choose which one to install and use.

    For example, for PyQt5, the default, you would use the following:
    ```sh
    pip install -e ".[pyqt,dev]"  # (quotes only needed for zsh shell)
    ```

    If you want to use PySide2 instead, you would use:
    ```sh
    pip install -e ".[pyside,dev]"  # (quotes only needed for zsh shell)
    ```

    Finally, if you already have a Qt backend installed or want to use an experimental one like Qt6 use:
    ```sh
    pip install -e ".[dev]"  # (quotes only needed for zsh shell)
    ```

    Note that in this last case you will need to install your Qt backend separately.

> 5. We use [`pre-commit`](https://pre-commit.com) to sort imports with
>    [`isort`](https://github.com/PyCQA/isort), format code with
>    [`black`](https://github.com/psf/black), and lint with
>    [`flake8`](https://github.com/PyCQA/flake8) automatically prior to each commit.

Reviewer 3: ```suggestion
5. We use [`pre-commit`](https://pre-commit.com) to format code with
   [`black`](https://github.com/psf/black) and lint with
   [`ruff`](https://github.com/charliermarsh/ruff) automatically prior to each commit.
```

Reviewer: @Czaki 
Should be:
We use pre-commit `to` format code with black`<delete COMMA>` and lint with ...`

Reviewer 3: @psobolewskiPhD fixed

You can then make a
[pull-request](https://docs.github.com/en/get-started/quickstart/contributing-to-projects#making-a-pull-request) to `napari`'s `main` branch.

## Contributing documentation

> If you wish to contribute documentation changes to napari, please read the [guide on contributing documentation](documentation/index.md). 

Reviewer: ```suggestion
If you wish to contribute to the napari documentation, please read the [guide on contributing documentation](documentation/index.md). 
```

You can then make a
[pull-request](https://docs.github.com/en/get-started/quickstart/contributing-to-projects#making-a-pull-request) to `napari`'s `main` branch.

> ## Contributing documentation

Reviewer 2: I would actually move this section up to just underneath setting up a `dev` installation. Pretty much everything else on this page is less important and less frequently used imo.

Author: Moved, thanks!
